File: Array/Palindrome.py
def palindrome(s):
	size = len(s)
	mid = size//2
	if size%2 == 0:
		left = mid - 1
		right = mid
	else:
		left = mid - 1
		right = mid + 1

	while left >= 0 and right < size:
		if s[left] != s[right]:
			return False
		left -= 1
		right += 1

	return True

print(palindrome("palindrome"))
print(palindrome("abccba"))
print(palindrome("abcba"))

File: Array/Palindrome.js
function palindrome(s){
	var mid = Math.floor(s.length/2);
	var left = mid - 1;
	var right = 0;
	if(s.length%2 == 0){
		right = mid;
	} else{
		right = mid + 1;
	}
	while(left >= 0 && right < s.length){
		if(s[left] !== s[right]){
			return false;
		}
		left--;
		right++;
	}
	return true;
}

console.log(palindrome("palindrome"));
console.log(palindrome("abccba"));
console.log(palindrome("abcba"));